src/server/graph-generation/metadataParser.js

```diff
@@ -14,118 +14,100 @@
 //Note : null is returned if nothing can be parsed
 function parse(subTree) {
   var result = [];
   var temp = [];
+  var databaseIDs = [];
 
   //Stop id subtree is invalid
   if (!(subTree)) return null;
 
   //Get the entity reference object
-  var eRef = searchForNode(subTree, 'bp:entityReference');
-
+  var eRef = treeTraversal.searchForExactNode(subTree, 'bp:entityReference');
 
   if (eRef) {
     //Get the standard name
-    result.push(searchOne(eRef, 'bp:standardName', 'Standard Name : '));
+    result.push(treeTraversal.searchOne(eRef, 'bp:standardName', 'Standard Name : '));
 
     //Get names
-    result.push(searchMultiple(eRef, 'bp:name', 'Names'));
+    result.push(treeTraversal.searchMultiple(eRef, 'bp:name', 'Names'));
 
     //Get database ids
-    result.push(searchMultiple(eRef, 'bp:xref', 'Database IDs'));
+    databaseIDs = treeTraversal.searchMultiple(eRef, 'bp:xref', 'Database IDs')
+    if (databaseIDs) databaseIDs = databaseIDs[1];
   }
 
   //Get data source
-  var source = searchForNode(subTree, 'bp:dataSource');
-  if (typeof source === 'string' ) result.push("Data Source : " + source);
-  else if(source) result.push("Data Source : " + source[0][1]);
+  var source = treeTraversal.searchForNode(subTree, 'bp:dataSource');
+  if (typeof source === 'string') result.push("Data Source : " + source);
+  else if (source) result.push("Data Source : " + treeTraversal.searchForNode(source, ''));
 
   //Get values if entity reference was not found
-  if (!eRef){
+  if (!eRef) {
     //Get BioPax Names
-    result.push(searchMultiple(subTree, 'bp:name', 'Names'));
+    result.push(treeTraversal.searchMultiple(subTree, 'bp:name', 'Names'));
 
     //Get Biopax database id's
-    result.push(["Database IDs", searchForNode(subTree, 'bp:xref')]);
+    databaseIDs = [treeTraversal.searchForNode(subTree, 'bp:xref')];
   }
 
-
   //Get cellular location
-  var location = searchForNode(subTree, 'bp:cellularLocation');
-  if(location) {
-    var term = searchForNode(location, 'bp:term');
-    result.push('Cellular Location : ' + term); 
+  var location = treeTraversal.searchForNode(subTree, 'bp:cellularLocation');
+  if (location) {
+    //var term = searchForNode(location, 'bp:term');
+    var term = treeTraversal.searchForFirst(location, 'bp:term');
+    if (term) result.push('Cellular Location : ' + term);
   }
 
   //Get all comments
-  result.push(searchMultiple(subTree, 'bp:comment', 'Comments')); 
+  result.push(treeTraversal.searchMultiple(subTree, 'bp:comment', 'Comments'));
+
+  //Get display name
+  temp = treeTraversal.searchForNode(subTree, 'bp:displayName');
+  if (temp) result.push('Display Name : ' + temp);
+
+  //Parse database id objects
+  if (databaseIDs) databaseIDs = parseDatabaseIDs(databaseIDs);
+  if (databaseIDs) result.push(['Database IDs', databaseIDs]);
 
   //Remove all invalid values
-  for (var i = 0; i < result.length; i++){
-    if(!(result[i])){
+  for (var i = 0; i < result.length; i++) {
```

Comment:
You have to iterate in reverse order if you remove from an array in-iteration.  Or you have to update the index on each iteration that modifies the array length.

Reply:
Parser

Reply:
This was mistakenly included in this PR. The newer version of the script uses .filter  to produce an array without the invalid values.

Reply:
All the latest code needs to be put up.  Moving forward, we need the code to be a reviewed, common base with small per-feature (or per-issue) PRs.